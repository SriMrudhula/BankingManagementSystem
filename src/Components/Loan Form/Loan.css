.err{
    font-size: 10px;
    color: red;
}
.img2{
    background-image: url('../../Asserts/banking.jpg');
    height: 590px;
    background-repeat: no-repeat;
    background-size: cover;
    margin-top: -90px;
    width:100%;
    background-attachment: fixed;
}

.b4{
    display:inline-block;
    width: 500px;
    height:600px;
    padding: 20px;
    background-color: rgba(0,0,0,0.5);
    background-size: 400px 480px;   
    margin-top:100px;
    position: relative;
     } 
     .b5{
        display:inline-block;
        width: 300px;
        height: 100px;
        padding: 50px;
        background-color: rgba(0,0,0,0.5);
        background-size: 400px 480px;  
        margin-left:100px;    
        margin-top:-100px;
        position: relative;
         } 
     label{
       font-size: large;
        color: white;
        margin-left: 30px;
    }
    input{
        margin-left: 60px;
    }
    .heading{
        color: white;
        font-size: 30px;
    }
    .btn1{
        margin-left: auto;
        height:50px;
    width:100px;
    border-radius: 10px;
    }
    .dis{
        display:none;
    }
    .dis1{
        display:block;
    }

    .container {
        width: 500px;
        clear: both;
      }
      
      .container input {
        width: 100%;
        clear: both;
      }