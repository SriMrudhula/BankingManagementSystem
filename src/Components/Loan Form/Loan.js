import React, { Component } from 'react';
import './Loan.css';
import AppBar from '@material-ui/core/AppBar';
import axios from 'axios';
import Login from '../Login Form/Login';
import { BrowserRouter, Route } from 'react-router-dom'

class Loan extends Component
{
  state={    
      id:"",
      accHoldername:"",
      loanType:"",
      loanAmount:"",
      loanApplyDate:"",
      rateOfInterest:"",
      durationOfLoan:"",
      courseFee:"",
      course:"",
      fatherName:"",
      fatherOcc:"",
      fathAnInc:"",
      annualInc:"",
      cmpName:"",
      desig:"",
      totExp:"",
      expForCurrentCmp:"",
      loan:false,
      loanValue:0
    }
    constructor(){
        super();
        this.state.id=localStorage.getItem("id");
        this.state.accHoldername= localStorage.getItem("name");
        console.log(this.state.accHoldername+"  "+this.state.id);
      }
    
    setLoanType =event => {
        this.setState({
            [event.target.name]:event.target.value
        });
        event=document.getElementById('loanType').value;
    
            if(event==="Student Loan"){
                this.state.loanValue=1;
            }
    else if(event==="Home/Personal Loan")
        this.state.loanValue=2;
    }
    hadLoan=()=>{
        axios.get("http://localhost:3000/loan?id="+this.state.id+"&name="+this.state.accHoldername)
        .then(response =>{
            console.log(response.data.length);
            if(response.data.length!==0){
               this.state.loan=true;
               if(response.data[0].loanValue===1)
               {
               alert("                 You have already taken " +response.data[0].loanType+
               "\n Loan Amount : "+response.data[0].loanAmount+
               "         Loan Apply Date : "+response.data[0].loanApplyDate+
               "\n Rate Of Interest : "+response.data[0].rateOfInterest+
               "                Duration Of Loan : "+response.data[0].durationOfLoan+
               "\n Course Fee : "+response.data[0].courseFee+
               "            Course : "+response.data[0].course+
               "\n Father Name : "+response.data[0].fatherName+
               "                Father Occupation:"+response.data[0].fatherOcc+
               "\n Father Annual Income:"+response.data[0].fathAnInc);
               }
            }
               else{
                   this.state.loan=false;
                   alert("You do not have any loan");
               
            }
        })
        .catch(error=>{
            console.log(error);
        })
    }
    handleChange =event=>{
        this.setState({
            [event.target.name]:event.target.value
        });
    };
    handleClick = event => {
        this.props.handleClick(event);     
    };   
    handleSubmit = event => {
        event.preventDefault();    
        console.log(this.state);
        this.hadLoan();
        if(this.state.loan===false){
        axios.post('http://localhost:3000/loan',this.state)
        .then(respone =>{
            console.log(respone)  
            alert("Applied for loan successfully");
            this.handleClick("home");  
        })
        .catch(error=>{
            console.log(error)
        })
    }
    };

    render()
    {
    return (
        <div className="img3">
            <div className="b4">
            <form onSubmit={this.handleSubmit} >
            <label> Name</label>
        <input type="text" id="name" value={this.state.accHoldername} name="name" onChange={this.handleChange} required/><br/><br/>       
        <label>Loan Type</label>
        <select  value={this.state.loanType} id="loanType" name="loanType" onChange={this.setLoanType} required>
            <option>Loan Type</option>
            <option value="Student Loan">Student Loan</option>
            <option value="Home/Personal Loan">Home/Personal Loan</option>
        </select><br/><br/>
        <label>Loan Amount</label>
        <input type="number" value={this.state.loanAmount} name="loanAmount" onChange={this.handleChange} required/>
        <br/><br/>
        <label>Loan Apply Date</label>
        <input type="date" value={this.state.loanApplyDate}  onChange={this.handleChange} name="loanApplyDate" required/>
        <br/><br/>
        <label>Rate of Interest</label>
        <input type="number"  value={this.state.rateOfInterest}  onChange={this.handleChange} name="rateOfInterest" required/>
        <br/><br/>
        <label>Duration of Time</label>
        <input type="number"  value={this.state.durationOfLoan}  onChange={this.handleChange} name="durationOfLoan" required/>
        <br/><br/>
        <div className={this.state.loanValue==1?"dis1":"dis"}>
                <label>Course Fee</label>
                <input type="number" value={this.state.courseFee}  name="courseFee" onChange={this.handleChange} />
                <br/><br/>
                <label>Course</label>
                <input type="text" value={this.state.course}  onChange={this.handleChange} name="course"/>
                <br/><br/>
                <label>Father Name</label>
                <input type="text"  value={this.state.fatherName}  onChange={this.handleChange} name="fatherName" />
                <br/><br/>
                <label>Father Occupation</label>
                <input type="text"  value={this.state.fatherOcc}  onChange={this.handleChange} name="fatherOcc" />
                <br/><br/>
                <label>Father Ann Income</label>
                <input type="text"  value={this.state.fathAnInc}  onChange={this.handleChange} name="fathAnInc" />
                <br/><br/>
                </div>
                <div className={this.state.loanValue==2?"dis1":"dis"}>
                     {/* Home Loan */}
        <label>Annual Income</label>
        <input type="number" value={this.state.annualInc} name="annualInc" onChange={this.handleChange}/>
        <br/><br/>
        <label>Company Name</label>
        <input type="text" value={this.state.cmpName}  onChange={this.handleChange} name="cmpName"/>
        <br/><br/>
        <label>Designation</label>
        <input type="text"  value={this.state.desig}  onChange={this.handleChange} name="desig"/>
        <br/><br/>
        <label>Total Experience</label>
        <input type="number"  value={this.state.totExp}  onChange={this.handleChange} name="totExp"/>
        <br/><br/>
        <label>Experience with Current Cmp</label>
        <input type="text"  value={this.state.expForCurrentCmp}  onChange={this.handleChange} name="expForCurrentCmp" />       <br/><br/>
        </div>
        <button className="btn1">Apply Loan</button>
        </form> 
        </div>
        <div className="b5">
            <div className="data">Check Present Loan</div><br/>
            <button className="btn1" onClick={this.hadLoan}>My Loan</button>
        </div>
        </div>
        )
    }
}

export default Loan;