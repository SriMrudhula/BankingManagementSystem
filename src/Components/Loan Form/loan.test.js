import * as React from 'react';
import Enzyme,{shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Loan from './Loan';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
Enzyme.configure({ adapter: new Adapter() });
describe('Test when form is submitted', () => {
    it('should submit form data', () => {
        localStorage.setItem("id",10);
        let form = shallow(<Loan/>); 
        form.instance().handleSubmit = jest.fn();
        form.instance().handleSubmit();
        form.update(); 
        form.find('#loansubmit').simulate('submit');
        expect(form.find('#loansubmit').length).toEqual(1);
        expect(form.instance().handleSubmit).toHaveBeenCalled();
    });
    it('renders correctly', () => {
        const tree = renderer.create(<BrowserRouter><Loan/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
      });
});
