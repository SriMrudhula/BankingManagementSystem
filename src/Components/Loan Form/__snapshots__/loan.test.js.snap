// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test when form is submitted renders correctly 1`] = `
<div
  className="img2"
>
  <div
    className="row"
  >
    <div
      className="b4"
    >
      <div
        className="heading"
      >
        Loan
      </div>
      <div
        className="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <input
                  className="form-control"
                  id="name"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  required={true}
                  type="text"
                  value={null}
                />
              </div>
              <div
                className="col"
              >
                <select
                  className="form-control"
                  id="loanType"
                  name="loanType"
                  onChange={[Function]}
                  required={true}
                  value=""
                >
                  <option>
                    Loan Type
                  </option>
                  <option
                    value="Student Loan"
                  >
                    Student Loan
                  </option>
                  <option
                    value="Home/Personal Loan"
                  >
                    Home/Personal Loan
                  </option>
                </select>
              </div>
            </div>
            <br />
            <div
              className="row"
            >
              <div
                className="col"
              >
                <input
                  className="form-control"
                  name="loanAmount"
                  onChange={[Function]}
                  placeholder="Loan Amount"
                  required={true}
                  type="number"
                  value=""
                />
              </div>
              <div
                className="col"
              >
                <input
                  className="form-control"
                  name="loanApplyDate"
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Loan Apply Date"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <br />
            <div
              className="row"
            >
              <div
                className="col"
              >
                <input
                  className="form-control"
                  name="rateOfInterest"
                  onChange={[Function]}
                  placeholder="Rate of Interest"
                  required={true}
                  type="number"
                  value=""
                />
              </div>
              <div
                className="col"
              >
                <input
                  className="form-control"
                  name="durationOfLoan"
                  onChange={[Function]}
                  placeholder="Duration of Time"
                  required={true}
                  type="number"
                  value=""
                />
              </div>
            </div>
            <br />
            <div
              className="dis"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <input
                    className="form-control"
                    name="courseFee"
                    onChange={[Function]}
                    placeholder="Course Fee"
                    type="number"
                    value=""
                  />
                </div>
                <div
                  className="col"
                >
                  <input
                    className="form-control"
                    name="course"
                    onChange={[Function]}
                    placeholder="Course"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <br />
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <input
                    className="form-control"
                    name="fatherName"
                    onChange={[Function]}
                    placeholder="Father Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="col"
                >
                  <input
                    className="form-control"
                    name="fatherOcc"
                    onChange={[Function]}
                    placeholder="Father Occupation"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="col"
                >
                  <input
                    className="form-control"
                    name="fathAnInc"
                    onChange={[Function]}
                    placeholder="Father Annual Income"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <br />
            </div>
            <div
              className="dis"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <input
                    className="form-control"
                    name="annualInc"
                    onChange={[Function]}
                    placeholder="Annual Income"
                    type="number"
                    value=""
                  />
                </div>
                <div
                  className="col"
                >
                  <input
                    className="form-control"
                    name="cmpName"
                    onChange={[Function]}
                    placeholder="Company Name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <br />
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <input
                    className="form-control"
                    name="desig"
                    onChange={[Function]}
                    placeholder="Designation"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="col"
                >
                  <input
                    className="form-control"
                    name="totExp"
                    onChange={[Function]}
                    placeholder="Total Experience"
                    type="number"
                    value=""
                  />
                </div>
                <div
                  className="col"
                >
                  <input
                    className="form-control"
                    name="expForCurrentCmp"
                    onChange={[Function]}
                    placeholder="Current Cmp Experience"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <br />
            </div>
            <button
              className="btn-primary"
              id="loansubmit"
            >
              Apply Loan
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      className="b5"
    >
      <button
        className="btn1"
        onClick={[Function]}
      >
        My Loan
      </button>
    </div>
  </div>
</div>
`;
